src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int buy = 0;
+        int supply = 0;
+        int result = 0;
+
+        try (BufferedReader reader =
+                     new BufferedReader(new FileReader(fromFileName))) { //try-with-resources
+
+            String line;
+
+            while ((line = reader.readLine()) != null) { // Читаємо файл, рядок за рядком
+
+                String[] parts = line.split(",");
+                String definition = parts[0];
+                int amountOfDefinition = Integer.parseInt(parts[1]);
+                if (definition.equals("buy")) {
+                    buy += amountOfDefinition;
+                }
+                if (definition.equals("supply")) {
+                    supply += amountOfDefinition;
+                }
+            }
+            result = supply - buy;
+
+            createReport(toFileName, buy, supply, result);
+
+        } catch (IOException e) {
+            e.printStackTrace(); // Виводимо стек виключення в разі помилки
+        }
+    }
+
+    private void createReport(String tofileName, int buy, int supply, int result) {
```

Comment:
The parameter name `tofileName` should be `toFileName` to maintain consistency with the naming convention used in the `getStatistic` method.